pocs/camera/camera.py

```diff
@@ -302,7 +302,13 @@ def take_observation(self, observation, headers=None, filename=None, **kwargs):
 
         # pop exptime from kwarg as its now in exptime
         exptime = kwargs.pop('exptime', observation.exptime.value)
-        exposure_event = self.take_exposure(seconds=exptime, filename=file_path, **kwargs)
+
+        # Move the filerwheel into position
+        if (self.filterwheel is not None) and (observation.filter_name is not None):
+            self.filterwheel.move_to(observation.filter_name)
+
```

Comment:
I think this should actually be pulled into `_setup_observation` rather than be directly placed in `take_observation`.

Related to that, it would be nice to populate the FITS headers during this PR as well, which happens in `_setup_observation`, so also seems to make sense.